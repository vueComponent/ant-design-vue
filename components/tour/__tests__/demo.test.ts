import demoTest from '../../../tests/shared/demoTest';

demoTest('tour');
