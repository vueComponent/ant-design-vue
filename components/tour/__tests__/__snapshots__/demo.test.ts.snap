// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ./components/tour/demo/basic.tsx correctly 1`] = `
Array [
  <button
    class="ant-btn ant-btn-primary"
    type="button"
  >
    <span>
      Begin Tour
    </span>
  </button>,
  <div
    class="ant-divider ant-divider-horizontal"
    role="separator"
  />,
  <div
    class="ant-space ant-space-horizontal ant-space-align-center"
  >
    <div
      class="ant-space-item"
      style="margin-right:8px"
    >
      <button
        class="ant-btn ant-btn-default"
        type="button"
      >
        <span>
           Upload
        </span>
      </button>
    </div>
    <div
      class="ant-space-item"
      style="margin-right:8px"
    >
      <button
        class="ant-btn ant-btn-primary"
        type="button"
      >
        <span>
          Save
        </span>
      </button>
    </div>
    <div
      class="ant-space-item"
    >
      <button
        class="ant-btn ant-btn-default ant-btn-icon-only"
        type="button"
      >
        <span
          aria-label="ellipsis"
          class="anticon anticon-ellipsis"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="ellipsis"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>,
]
`;

exports[`renders ./components/tour/demo/non-modal.tsx correctly 1`] = `
Array [
  <button
    class="ant-btn ant-btn-primary"
    type="button"
  >
    <span>
      Begin non-modal Tour
    </span>
  </button>,
  <div
    class="ant-divider ant-divider-horizontal"
    role="separator"
  />,
  <div
    class="ant-space ant-space-horizontal ant-space-align-center"
  >
    <div
      class="ant-space-item"
      style="margin-right:8px"
    >
      <button
        class="ant-btn ant-btn-default"
        type="button"
      >
        <span>
           Upload
        </span>
      </button>
    </div>
    <div
      class="ant-space-item"
      style="margin-right:8px"
    >
      <button
        class="ant-btn ant-btn-primary"
        type="button"
      >
        <span>
          Save
        </span>
      </button>
    </div>
    <div
      class="ant-space-item"
    >
      <button
        class="ant-btn ant-btn-default ant-btn-icon-only"
        type="button"
      >
        <span
          aria-label="ellipsis"
          class="anticon anticon-ellipsis"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="ellipsis"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>,
]
`;

exports[`renders ./components/tour/demo/placement.tsx correctly 1`] = `
<button
  class="ant-btn ant-btn-primary"
  type="button"
>
  <span>
    Begin Tour
  </span>
</button>
`;

exports[`renders ./components/tour/demo/render-panel.tsx correctly 1`] = `
<div
  style="display:flex;flex-direction:column;row-gap:16px;background:rgba(50,0,0,0.65);padding:8px"
>
  <div
    class="ant-tour ant-tour-pure ant-tour-placement-top ant-tour-pure"
  >
    <div
      class="ant-tour-content"
    >
      <div
        class="ant-tour-arrow"
      />
      <div
        class="ant-tour-inner"
        role="tooltip"
      >
        <div
          class="ant-tour-content"
        >
          <div
            class="ant-tour-inner"
          >
            <span
              aria-label="close"
              class="anticon anticon-close ant-tour-close"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="close"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                />
              </svg>
            </span>
            <div
              class="ant-tour-header"
            >
              <div
                class="ant-tour-title"
              >
                Hello World!
              </div>
            </div>
            <div
              class="ant-tour-description"
            >
              Hello World?!
            </div>
            <div
              class="ant-tour-footer"
            >
              <div
                class="ant-tour-sliders"
              >
                <span
                  class="ant-tour-slider-active ant-tour-slider"
                />
                <span
                  class="ant-tour-slider"
                />
                <span
                  class="ant-tour-slider"
                />
                <span
                  class="ant-tour-slider"
                />
                <span
                  class="ant-tour-slider"
                />
                <span
                  class="ant-tour-slider"
                />
              </div>
              <div
                class="ant-tour-buttons"
              >
                <button
                  class="ant-btn ant-btn-primary ant-btn-sm ant-tour-next-btn"
                  type="button"
                >
                  <span>
                    Next
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-tour ant-tour-pure ant-tour-placement-top ant-tour-pure"
  >
    <div
      class="ant-tour-content"
    >
      <div
        class="ant-tour-arrow"
      />
      <div
        class="ant-tour-inner"
        role="tooltip"
      >
        <div
          class="ant-tour-content"
        >
          <div
            class="ant-tour-inner"
          >
            <span
              aria-label="close"
              class="anticon anticon-close ant-tour-close"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="close"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                />
              </svg>
            </span>
            <div
              class="ant-tour-cover"
            >
              <img
                alt="tour.png"
                src="https://user-images.githubusercontent.com/5378891/197385811-55df8480-7ff4-44bd-9d43-a7dade598d70.png"
              />
            </div>
            <div
              class="ant-tour-header"
            >
              <div
                class="ant-tour-title"
              >
                Hello World!
              </div>
            </div>
            <div
              class="ant-tour-description"
            >
              Hello World?!
            </div>
            <div
              class="ant-tour-footer"
            >
              <div
                class="ant-tour-sliders"
              >
                <span
                  class="ant-tour-slider"
                />
                <span
                  class="ant-tour-slider"
                />
                <span
                  class="ant-tour-slider"
                />
                <span
                  class="ant-tour-slider"
                />
                <span
                  class="ant-tour-slider"
                />
                <span
                  class="ant-tour-slider-active ant-tour-slider"
                />
                <span
                  class="ant-tour-slider"
                />
              </div>
              <div
                class="ant-tour-buttons"
              >
                <button
                  class="ant-btn ant-btn-default ant-btn-sm ant-tour-prev-btn"
                  type="button"
                >
                  <span>
                    Previous
                  </span>
                </button>
                <button
                  class="ant-btn ant-btn-primary ant-btn-sm ant-tour-next-btn"
                  type="button"
                >
                  <span>
                    Next
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-tour ant-tour-pure ant-tour-placement-top ant-tour-pure ant-tour-primary"
  >
    <div
      class="ant-tour-content"
    >
      <div
        class="ant-tour-arrow"
      />
      <div
        class="ant-tour-inner"
        role="tooltip"
      >
        <div
          class="ant-tour-primary ant-tour-content"
        >
          <div
            class="ant-tour-inner"
          >
            <span
              aria-label="close"
              class="anticon anticon-close ant-tour-close"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="close"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                />
              </svg>
            </span>
            <div
              class="ant-tour-header"
            >
              <div
                class="ant-tour-title"
              >
                Hello World!
              </div>
            </div>
            <div
              class="ant-tour-description"
            >
              Hello World?!
            </div>
            <div
              class="ant-tour-footer"
            >
              <div
                class="ant-tour-sliders"
              >
                <span
                  class="ant-tour-slider"
                />
                <span
                  class="ant-tour-slider"
                />
                <span
                  class="ant-tour-slider"
                />
                <span
                  class="ant-tour-slider"
                />
                <span
                  class="ant-tour-slider-active ant-tour-slider"
                />
              </div>
              <div
                class="ant-tour-buttons"
              >
                <button
                  class="ant-btn ant-btn-default ant-btn-sm ant-btn-background-ghost ant-tour-prev-btn"
                  type="button"
                >
                  <span>
                    Previous
                  </span>
                </button>
                <button
                  class="ant-btn ant-btn-default ant-btn-sm ant-tour-next-btn"
                  type="button"
                >
                  <span>
                    Finish
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
